bundles/framework/announcements/view/AnnouncementsCollapse.jsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import PropTypes from 'prop-types';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import { Message, Collapse, CollapsePanel } from 'oskari-ui';
+import { AnnouncementsModal } from './AnnouncementsModal';
+import { Divider } from 'antd';
+
+//Collapse panel -> set title, content and date range according to announcement
+
+const { Panel } = Collapse;
+
+const AnnouncementsCollapse = ({controller, updated, panels, modals, checked }) => {
+
+      if (!updated) {
+        controller.getAnnouncements(function(ann){
+          console.log("get announcements collapse");
+          console.log(ann);
+
+        
+          var panels = [];
+          var modals = [];
+            ann.data.forEach((announcement) => {
+
+              if (announcement.active && controller.showModal(announcement.id)) {
+                //if announcement is active, then show pop-up of the content
+                modals.push(announcement);
+              }
+              panels.push(
+                      <CollapsePanel header={announcement.title} key={announcement.id}>
+                          <h3>{announcement.title}</h3>
+                          <p>{announcement.content}</p>
+                          <Divider />
+                          <b><Message messageKey={'valid'} /></b>
+                          <p>{announcement.begin_date} - {announcement.end_date}</p>
+                      </CollapsePanel>
+                );
+            });
+            controller.updatePanelsModals(panels,modals);
+          });
+      }
```

Comment:
See the suggestion on the handler for fetching announcements at start and not on render. Refactor so you don't have to push modals and panels to the state, but map them on the render from announcements.